test/blas1_test.hpp

```diff
@@ -0,0 +1,175 @@
+#ifndef BLAS1_TEST_HPP_DFQO1OHP
+#define BLAS1_TEST_HPP_DFQO1OHP
+
+#include <cmath>
+#include <complex>
+#include <cstdlib>
+#include <iomanip>
+#include <iostream>
+#include <vector>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <interface/blas1_interface_sycl.hpp>
+
+using namespace cl::sycl;
+using namespace blas;
+
+template <typename ClassName>
+struct option_size;
+#define RANDOM_SIZE UINT_MAX
+#define RANDOM_STRD UINT_MAX
```

Comment:
To avoid compilation errors I thought it would be cheaper to make it into macros than to split into .hpp and .cpp file to avoid symbol duplication in case this header is included in more than one file.